akka-actor/src/main/scala/akka/io/dns/internal/AsyncDnsResolver.scala

```diff
@@ -37,8 +38,6 @@ private[io] final class AsyncDnsResolver(
 
   import AsyncDnsResolver._
 
-  implicit val ec: ExecutionContextExecutor = context.dispatcher
```

Comment:
I do kind of feel that having an implicit execution context in an actor is a bad sign, by decreasing the cognitive cost of adding a future callback within the actor.